### PR TITLE
Expiration summary email for security team

### PR DESCRIPTION
**Pull request by jtschladen**
Adds support for sending a daily summary email of expiring certs to the security team. Sample:

Screen Shot 2020-12-08 at 11 47 12 AM

### CODE REVIEW

**Review by hosseinsh — COMMENTED** [review]
Thanks for the PR @jtschladen.

I was wondering if there would be value in listing all expiring certs in the coming X days?
and then make sure to only send it during workdays?

This way security would get reminded about certs that are still about to expire, and not action has been taken on them. 


The list certs could then be sorted based on the expiration date. 

Also another improvement could b dividing the certs among attached to endpoint and not attached to anything. 

These can be follow-up improvements.

File: lemur/common/celery.py [hunk 1]
```diff
@@ -857,6 +857,42 @@ def notify_authority_expirations():
...
+    }
+
+    if task_id and is_task_active(function, task_id, None):
+        log_data["message"] = "Skipping task: Task is already active"
+        current_app.logger.debug(log_data)
+        return
+
+    current_app.logger.debug(log_data)
+    try:
+        cli_notification.send_security_expiration_summary()
```
[the remark this answers is not in the document]
Comment (jtschladen) [reply lemur/common/celery.py:884]: Thanks, you're correct!

File: lemur/notifications/messaging.py [hunk 2]
```diff
@@ -63,6 +63,39 @@ def get_certificates(exclude=None):
...
+    if exclude:
+        for e in exclude:
+            exclude_conditions.append(~Certificate.name.ilike("%{}%".format(e)))
+
+        q = q.filter(and_(*exclude_conditions))
+
+    certs = []
+    for c in windowed_query(q, Certificate.id, 10000):
+        days_remaining = (c.not_after - now).days
+        if days_remaining in expiration_summary_intervals:
```
[the remark this answers is not in the document]
Comment (jtschladen) [reply lemur/notifications/messaging.py:94]: Makes sense to me. I've made the change, but I'm curious whether you think the "expiring in x days" headers are helpful or useless at this point? The expiration dates are embedded in the cert info, and I think I could probably sort the certs by expiry date, but perhaps the headers make it more obvious at a glance how urgent each one is.

Screen Shot 2020-12-08 at 6 23 16 PM

**Comment by jtschladen** [comment]
> and then make sure to only send it during workdays?

I wonder if it would be more open-ended not to enforce this in code, but rather to rely on the cron/celery configuration to handle only calling this on weekdays? That way, if someone has a different work week than M-F (not too likely) or they just want it sent every day, they have that option.

> Also another improvement could be dividing the certs among attached to endpoint and not attached to anything.

I'll take this down as a follow-up.

File: lemur/notifications/messaging.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hosseinsh) [reply lemur/notifications/messaging.py]: I think headers would do the trick, to bucket the certs per day.

**Comment by hosseinsh** [comment]
Excellent point with using the scheduled cron or celery task to define the sending timeline.

With respect to dividing between endpoint/no endpoint, the initial test phase would provide us better data to understand how helpful this would be.

**Review by hosseinsh — APPROVED** [review]
Awesome work!

**Pull request merged**